### PR TITLE
feat: Support faker modifiers on Fabricator

### LINKED ISSUE #6812: FabricatorLiveTest::testCreateAddsCountToDatabase failed
```
There was 1 error:

1) CodeIgniter\Database\Live\FabricatorLiveTest::testCreateAddsCountToDatabase
CodeIgniter\Database\Exceptions\DatabaseException: [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Violation of UNIQUE KEY constraint 'db_user_email'. Cannot insert duplicate key in object 'dbo.db_user'. The duplicate key value is (rrunte@yahoo.com).

/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/BaseConnection.php:632
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/BaseBuilder.php:2196
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Model.php:305
/home/runner/work/CodeIgniter4/CodeIgniter4/system/BaseModel.php:781
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Model.php:694
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Test/Fabricator.php:469
/home/runner/work/CodeIgniter4/CodeIgniter4/tests/system/Database/Live/FabricatorLiveTest.php:53
phpvfscomposer:///home/runner/work/CodeIgniter4/CodeIgniter4/vendor/phpunit/phpunit/phpunit:97

Caused by
CodeIgniter\Database\Exceptions\DatabaseException: [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Violation of UNIQUE KEY constraint 'db_user_email'. Cannot insert duplicate key in object 'dbo.db_user'. The duplicate key value is (rrunte@yahoo.com).

/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/SQLSRV/Connection.php:478
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/BaseConnection.php:678
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/BaseConnection.php:600
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/BaseBuilder.php:2196
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Model.php:305
/home/runner/work/CodeIgniter4/CodeIgniter4/system/BaseModel.php:781
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Model.php:694
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Test/Fabricator.php:469
/home/runner/work/CodeIgniter4/CodeIgniter4/tests/system/Database/Live/FabricatorLiveTest.php:53
phpvfscomposer:///home/runner/work/CodeIgniter4/CodeIgniter4/vendor/phpunit/phpunit/phpunit:97

--
```
https://github.com/codeigniter4/CodeIgniter4/actions/runs/3390547385/jobs/5634814765

### PR DESCRIPTION
**Pull request by paulbalandan**
**Description**
Closes #6812 

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [x] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Review by michalsn — APPROVED** [review]
Nice feature.

File: user_guide_src/source/testing/fabricator.rst [hunk 1]
```diff
@@ -62,6 +62,23 @@ a child class in your test support folder:
...
+=================
+
+Faker provides three special providers, ``unique()``, ``optional()``, and ``valid()``,
+to be called before any provider. Fabricator fully supports these modifiers by providing
+dedicated methods.
+
+.. literalinclude:: fabricator/022.php
+
+The arguments passed after the field name are passed directly to the modifiers as-is. You can refer
+to Faker's documentation on these modifiers for details.
```
Comment (kenjis) [inline user_guide_src/source/testing/fabricator.rst:75]:
```suggestion
to `Faker's documentation on modifiers`_ for details.

.. _Faker's documentation on modifiers: https://fakerphp.github.io/#modifiers
```

**Comment by kenjis** [comment]
@paulbalandan Add a short description in
https://github.com/codeigniter4/CodeIgniter4/blob/4.5/user_guide_src/source/changelogs/v4.5.0.rst#testing

**Comment by kenjis** [comment]
Can you rebase to check GA checks?

**Review by kenjis — APPROVED** [review]
LGTM!

**Pull request merged**